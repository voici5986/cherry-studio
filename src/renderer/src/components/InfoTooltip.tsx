import { Tooltip, TooltipProps } from 'antd'
import { Info } from 'lucide-react'

type InheritedTooltipProps = Omit<TooltipProps, 'children'>

interface InfoTooltipProps extends InheritedTooltipProps {
  iconColor?: string
  iconSize?: string | number
  iconStyle?: React.CSSProperties
}

const InfoTooltip = ({ iconColor = 'var(--color-text-3)', iconSize = 14, iconStyle, ...rest }: InfoTooltipProps) => {
  return (
    <Tooltip {...rest}>
      <Info size={iconSize} color={iconColor} style={{ ...iconStyle }} role="img" aria-label="Information" />
    </Tooltip>
  )
}

export default InfoTooltip
