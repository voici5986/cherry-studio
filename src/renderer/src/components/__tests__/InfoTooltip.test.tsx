import { render, screen } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'

import InfoTooltip from '../InfoTooltip'

vi.mock('antd', () => ({
  Tooltip: ({ children, title }: { children: React.ReactNode; title: string }) => (
    <div>
      {children}
      {title && <div>{title}</div>}
    </div>
  )
}))

vi.mock('lucide-react', () => ({
  Info: ({ ref, ...props }) => (
    <div {...props} ref={ref} role="img" aria-label="Information">
      Info
    </div>
  )
}))

describe('InfoTooltip', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <InfoTooltip title="Test tooltip" placement="top" iconColor="#1890ff" iconStyle={{ fontSize: '16px' }} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should pass title prop to the underlying Tooltip component', () => {
    const tooltipText = 'This is helpful information'
    render(<InfoTooltip title={tooltipText} />)

    expect(screen.getByRole('img', { name: 'Information' })).toBeInTheDocument()
    expect(screen.getByText(tooltipText)).toBeInTheDocument()
  })
})
