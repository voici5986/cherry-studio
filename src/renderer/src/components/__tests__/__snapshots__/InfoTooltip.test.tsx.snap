// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoTooltip > should match snapshot 1`] = `
<div>
  <div
    aria-label="Information"
    color="#1890ff"
    role="img"
    size="14"
    style="font-size: 16px;"
  >
    Info
  </div>
  <div>
    Test tooltip
  </div>
</div>
`;
